Updating Modern Web Apps eBook to 3.1

ardalis opened the pull request:
## Summary

Updating content and code samples to ASP.NET Core 3.1.

ardalis committed:
Ignoring github workflow

ardalis committed:
rolling back change

ardalis committed:
Updating first 6 chapters

ardalis committed:
updated chapter 7

ardalis committed:
updating chapters 8,9

ardalis committed:
updating links and vue notes

ardalis committed:
update screenshot

ardalis committed:
updating for 3.1

ardalis committed:
fix references to UseMvc

ardalis committed:
update date

ardalis committed:
updating test code to match 3.1 version of eShopOnWeb

nishanil commented on docs/architecture/modern-web-apps-azure/choose-between-traditional-web-and-single-page-apps.md:
```diff
@@ -29,13 +29,23 @@ You should use a SPA when:
 - Your application must already expose an API for other (internal or public) clients.
 
-Additionally, SPA frameworks require greater architectural and security expertise. They experience greater churn due to frequent updates and new frameworks than traditional web applications. Configuring automated build and deployment processes and utilizing deployment options like containers are more difficult with SPA applications than traditional web apps.
+Additionally, SPA frameworks require greater architectural and security expertise. They experience greater churn due to frequent updates and new frameworks than traditional web applications. Configuring automated build and deployment processes and utilizing deployment options like containers mahy be more difficult with SPA applications than traditional web apps.
```
```suggestion
Additionally, SPA frameworks require greater architectural and security expertise. They experience greater churn due to frequent updates and new frameworks than traditional web applications. Configuring automated build and deployment processes and utilizing deployment options like containers may be more difficult with SPA applications than traditional web apps.
```

nishanil commented on docs/architecture/modern-web-apps-azure/common-client-side-web-technologies.md:
```diff
@@ -3,7 +3,7 @@ title: Common client side web technologies
-ms.date: 01/30/2019
+ms.date: 12/4/2019
 ---
 # Common client side web technologies
```
Some Acrolinx errors found. I am unable to get to the exact statement because those existed prior to your change. Can you fix these:
- replace all occurrences of `client side` with `client-side`
- `generally recommended` to just `recommended`
-  `a lot of` is used multiple times. Probably, replace with `numerous`, `many` or `much`
- `Are new projects building built with it?` remove `building`
- replace `well-documented` with `well documented`

nishanil commented on docs/architecture/modern-web-apps-azure/work-with-data-in-asp-net-core-apps.md:
```diff
@@ -194,9 +194,9 @@ private void ConfigureOrder(EntityTypeBuilder<Order> builder)
+In this example, the `ShipToAddress` property is of type `Address`. `Address` is a value object with several properties such as `Street` and `City`. EF Core maps the `Order` object to its table with one column per `Address` property, prefixing each column name with the name of the property. In this example, the `Order` table would include columns such as `ShipToAddress_Street` and `ShipToAddress_ ...
 
-[EF Core 2.2 introduces support for collections of owned entities](https://docs.microsoft.com/ef/core/what-is-new/ef-core-2.2#collections-of-owned-entities)
+Learn more about owned [entity support in EF Core](https://docs.microsoft.com/en-us/ef/core/modeling/owned-entities).
```
```suggestion
Learn more about owned [entity support in EF Core](https://docs.microsoft.com/ef/core/modeling/owned-entities).
```

mairaw replied:
I changed your link to have them supported in VS offline books too across books

nishanil reviewed (changes requested):
@ardalis Left a few comments for you to address. 
Can you also fix this page? http://docs.microsoft.com/en-us/dotnet/architecture/modern-web-apps-azure/development-process-for-azure
Replace Visual Studio 2017 references with Visual Studio 2019 (both screenshots and text).

Also, Update `Copyright © 2019` to `Copyright © 2020` on index page.

nishanil commented:
@mairaw I have reviewed the content from my side. If you have anything specific to docs, please feel free to add them here.

mairaw commented on docs/architecture/modern-web-apps-azure/architectural-principles.md:
```diff
@@ -3,7 +3,7 @@ title: Architectural principles
 author: ardalis
 ms.author: wiwagn
-ms.date: 02/16/2019
+ms.date: 12/4/2019
```
date format is mm/dd/yyyy 

```suggestion
ms.date: 12/04/2019
```

mairaw commented on docs/architecture/modern-web-apps-azure/choose-between-traditional-web-and-single-page-apps.md:
```diff
@@ -3,7 +3,7 @@ title: Choose between traditional web apps and single page apps
 author: ardalis
 ms.author: wiwagn
-ms.date: 01/30/2019
+ms.date: 12/4/2019
```
```suggestion
ms.date: 12/04/2019
```

mairaw commented on docs/architecture/modern-web-apps-azure/common-client-side-web-technologies.md:
```diff
@@ -3,7 +3,7 @@ title: Common client side web technologies
 author: ardalis
 ms.author: wiwagn
-ms.date: 01/30/2019
+ms.date: 12/4/2019
```
```suggestion
ms.date: 12/04/2019
```

mairaw commented on docs/architecture/modern-web-apps-azure/common-client-side-web-technologies.md:
```diff
@@ -14,6 +14,8 @@ ASP.NET Core applications are web applications and they typically rely on client
 
 While HTML and CSS are relatively stable, JavaScript, by means of the application frameworks and utilities developers work with to build web-based applications, is evolving at breakneck speed. This chapter looks at a few ways JavaScript is used by web developers as part of developing applications, as provides a high-level overview of the Angular and React client side libraries.
 
+**Note:** Blazor provides an alternative to JavaScript frameworks for building rich, interactive client user interfaces. Client-side Blazor support is still in preview so for now it is out of scope for this chapter.
```
```suggestion
> [!NOTE]
> Blazor provides an alternative to JavaScript frameworks for building rich, interactive client user interfaces. Client-side Blazor support is still in preview, so for now it's out of scope for this chapter.
```

mairaw commented on docs/architecture/modern-web-apps-azure/common-web-application-architectures.md:
```diff
@@ -3,7 +3,7 @@ title: Common web application architectures
 author: ardalis
 ms.author: wiwagn
-ms.date: 01/30/2019
+ms.date: 12/4/2019
```
```suggestion
ms.date: 12/04/2019
```

nishanil commented on docs/architecture/modern-web-apps-azure/index.md:
```diff
@@ -3,7 +3,7 @@ title: Architect modern web applications with ASP.NET Core and Azure
+ms.date: 12/4/2019
 ---
 
 # Architect Modern Web Applications with ASP.NET Core and Azure
```
Update `Copyright © 2019` to `Copyright © 2020` on this page.

mairaw commented on docs/architecture/modern-web-apps-azure/common-web-application-architectures.md:
```diff
@@ -93,8 +93,7 @@ The simplest approach to scaling a web application in Azure is to configure scal
 
-> [!NOTE]
-> The term Clean Architecture can be applied to applications that are built using DDD Principles as well as to those that are not built using DDD. In the case of the former, this combination may be referred to as "Clean DDD Architecture".
+The eShopOnWeb reference application uses the Clean Architecture approach in organizing its code into projects. You can find a solution template you can use as a staring point for your own ASP.NET Core on GitHub at [github.com/ardalis/cleanarchitecture](https://github.com/ardalis/cleanarchitecture).
```
Shouldn't this be on an official dotnet repo?

```suggestion
The eShopOnWeb reference application uses the Clean Architecture approach in organizing its code into projects. You can find a solution template you can use as a starting point for your own ASP.NET Core on the [ardalis/cleanarchitecture](https://github.com/ardalis/cleanarchitecture) GitHub repository.
```

ardalis replied:
MS doesn't have one - this one is mine and I'm referring folks to it as the author of the ebook. Hopefully that's still OK.

mairaw replied:
yep, it's fine. It's just that I've seen in the past work that was done as part of the e-book project pointing to someone's repo. Still not a problem, but those should be maintained by Microsoft.

mairaw commented on docs/architecture/modern-web-apps-azure/develop-asp-net-core-mvc-apps.md:
```diff
@@ -3,7 +3,7 @@ title: Developing ASP.NET Core MVC apps
 author: ardalis
 ms.author: wiwagn
-ms.date: 01/30/2019
+ms.date: 12/4/2019
```
```suggestion
ms.date: 12/04/2019
```

mairaw commented on docs/architecture/modern-web-apps-azure/modern-web-applications-characteristics.md:
```diff
@@ -3,7 +3,7 @@ title: Characteristics of modern web applications
 author: ardalis
 ms.author: wiwagn
-ms.date: 01/30/2019
+ms.date: 12/4/2019
```
```suggestion
ms.date: 12/04/2019
```

mairaw commented on docs/architecture/modern-web-apps-azure/test-asp-net-core-mvc-apps.md:
```diff
@@ -3,7 +3,7 @@ title: Test ASP.NET Core MVC apps
 author: ardalis
 ms.author: wiwagn
-ms.date: 01/30/2019
+ms.date: 12/4/2019
```
```suggestion
ms.date: 12/04/2019
```

mairaw commented on docs/architecture/modern-web-apps-azure/work-with-data-in-asp-net-core-apps.md:
```diff
@@ -3,7 +3,7 @@ title: Work with data in ASP.NET Core Apps
 author: ardalis
 ms.author: wiwagn
-ms.date: 01/30/2019
+ms.date: 12/4/2019
```
```suggestion
ms.date: 12/04/2019
```

mairaw commented on .gitignore:
```diff
@@ -40,3 +40,4 @@ ehthumbs_vista.db
 
 # Windows shortcuts
 *.lnk
+
```
```suggestion
```

mairaw commented on docs/architecture/modern-web-apps-azure/common-client-side-web-technologies.md:
```diff
@@ -14,6 +14,8 @@ ASP.NET Core applications are web applications and they typically rely on client
+
 ## HTML
 
 HTML (HyperText Markup Language) is the standard markup language used to create web pages and web applications. Its elements form the building blocks of pages, representing formatted text, images, form inputs, and other structures. When a browser makes a request to a URL, whether fetching a page or an application, the first thing that is returned is an HTML document. This HTML document may refere ...
```
we don't need to spell out well known acronyms like HTML

```suggestion
HTML is the standard markup language used to create web pages and web applications. Its elements form the building blocks of pages, representing formatted text, images, form inputs, and other structures. When a browser makes a request to a URL, whether fetching a page or an application, the first thing that is returned is an HTML document. This HTML document may reference or include additional information about its look and layout in the form of CSS, or behavior in the form of JavaScript.
```

mairaw commented on docs/architecture/modern-web-apps-azure/common-web-application-architectures.md:
```diff
@@ -93,8 +93,7 @@ The simplest approach to scaling a web application in Azure is to configure scal
 
```
line 37 - remove extra comma in "This follows the separation of concerns principle, and can"

mairaw commented on docs/architecture/modern-web-apps-azure/develop-asp-net-core-mvc-apps.md:
```diff
@@ -553,7 +544,7 @@ There are a few steps involved in the process of deploying your ASP.NET Core app
+In addition to a process manager, ASP.NET Core applications may use a reverse proxy server. A reverse proxy server receives HTTP requests from the Internet and forwards them to Kestrel after some preliminary handling. Reverse proxy servers provide a layer of security for the application. Kestrel also doesn't support hosting multiple applications on the same port, so techniques like host headers c ...
 
 ![Kestrel to Internet](./media/image7-5.png)
 
```
this article mentions MSDN article but it should be MSDN Magazine article

mairaw reviewed (commented):
A few comments

ardalis committed:
Apply suggestions from code review

Co-Authored-By: Maira Wenzel <mairaw@microsoft.com>
Co-Authored-By: Nish Anil <nish@microsoft.com>

ardalis committed:
address acrolinx errors

ardalis committed:
Responding to feedback; expanding on Blazor

ardalis committed:
remove comma

nishanil commented:
@ardalis There are conflicts in the branch.

ardalis committed:
Merge branch 'master' into ardalis/modern-web-apps-3_1

ardalis commented:
Should be good to go except for the cover image, which I haven't had a chance to play with.

nishanil commented:
@ardalis Looks good! Can you add in the cover and resolve conflicts so we can merge in?
@mairaw Are you good with the changes?

ardalis committed:
updating cover image

ardalis committed:
Merge branch 'ardalis/modern-web-apps-3_1' of https://github.com/ardalis/docs-1 into ardalis/modern-web-apps-3_1

ardalis committed:
fix linter and cover image

ardalis committed:
Merge branch 'master' into ardalis/modern-web-apps-3_1

ardalis commented:
:shipit: 
Needs final review and merge.

nishanil commented:
@ardalis Just noticed. The page `development-process-for-azure.md` has VS 2017 images and text references. Can you update them to Visual Studio 2019, please?

ardalis committed:
update vs2017 references to vs2019

ardalis commented:
@nishanil should be good now. I removed the image with vs2017 as it wasn't needed. I updated the figure numbers in text (but not the image file names). I also fixed one image that had a transparent background to make it white so it will look right in dark theme in the docs site.

ardalis committed:
Cleaning up two images to fix #15721

ardalis commented:
:shipit:

nishanil reviewed (approved):
LGTM
